[RUM-11519] Add RUM Action Tracking configuration options to the Babel Plugin (#968)
DataDog/dd-sdk-reactnative: packages/react-native-babel-plugin/README.md

Add the plugin to your Babel configuration. Depending on your setup, you might be using a `babel.config.js`, `.babelrc`, or similar.

> <b>Example configuration:</b>

Reviewer: ```suggestion
**Example configuration:**
```

};
```

### Configuration Options

You can configure the plugin to adjust how it processes your code, giving you control over its behavior and allowing you to tailor it to your project’s needs.

#### Top-level options

| Option               | Type   | Default | Description |
|-----------------------|--------|---------|-------------|
| `actionNameAttribute` | string | –       | The chosen attribute name to use for action names. |
| `components`          | object | –       | Component tracking configuration. |

---

#### `components` options

| Option          | Type    | Default | Description |
|-----------------|---------|---------|-------------|

> | `useContent`    | boolean | true   | Whether to use component content (e.g., children, props) as the action name. |

Reviewer: ```suggestion
| `useContent`    | boolean | true   | Whether to use component content (for example: children, props) as the action name. |
```

| `useNamePrefix` | boolean | true   | Whether to prefix actions with the component name. |
| `tracked`       | array   | –       | List of component-specific tracking configs. |

---

#### `components.tracked[]` (per component)

Each entry in the `tracked` array is an object with the following shape:

| Option          | Type    | Default              | Description |
|-----------------|---------|----------------------|-------------|
| `name`          | string  | –                    | The component name to track (e.g., `Button`). |
| `useContent`    | boolean | inherits from global | Override `useContent` for this component. |
| `useNamePrefix` | boolean | inherits from global | Override `useNamePrefix` for this component. |

> | `contentProp`   | string  | –                    | Property name to use for content instead of children (e.g., `"subTitle"`). |

Reviewer: ```suggestion
| `contentProp`   | string  | –                    | Property name to use for content instead of children (for example, `"subTitle"`). |
```

| `handlers`      | array   | –                    | List of event/action pairs to track. |

---

#### `components.tracked[].handlers[]`

| Field   | Type   | Description |
|---------|--------|-------------|

> | `event` | string | The event name to intercept (e.g., `"onPress"`). |

Reviewer: ```suggestion
| `event` | string | The event name to intercept (such as `"onPress"`). |
```

> | `action`| string | The RUM action name to associate with this event. <i>(Only `"TAP"` actions are currently supported)<i> |

Reviewer: ```suggestion
| `action`| string | The RUM action name to associate with this event. _(Only `"TAP"` actions are currently supported)_ |
```

---

> <b>Example configuration (<i>using configuration options</i>):</b>

Reviewer: ```suggestion
**Example configuration (_using configuration options_):**
```
